### PR TITLE
Feat/add tour form

### LINKED ISSUE #24: UI MVP: Add A Tour
I've opened this issue back up to track the next two pages of adding tour. Sorry for the confusion about what this issue included.

### PR DESCRIPTION
**Pull request by deadbelly**
### What Type of PR Is This?
- [x] New Feature 
- [ ] Bug Fix
- [ ] Testing
- [ ] Refactoring/Clean Up 

#### What does this branch do? 
This branch adds the initial +Tour screen as a controlled form. It does this with several re-usable components that we can tap to fill out the other two thirds of the form. These components include:

- `TextField.tsx` Which is a text input field connected to the main Form state. 
- `SectionTitle.tsx` Which includes a title for a section of input fields as well as a checkbox that automatically toggles when a user has completed that section.

#### Have you tested your branch? How?
I've rendered this component in `App.js` just so it's visible while we work on it. Eventually it should be conditionally rendered probably with a route. 

#### Do all existing unit and integration tests pass? 
N/A
#### Does this branch require special installation steps? 
No
#### Related Issues:
closes #24

### CODE REVIEW

File: src/components/Form/SectionTitle.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+
+interface SectionTitleProps {
+  title: string,
+  fields: string[],
+  isChecked: (fields: string[]) => boolean
+}
+
+export const SectionTitle: React.FC<SectionTitleProps> = ({title, fields, isChecked}) => {
```
Comment (tashiad) [inline src/components/Form/SectionTitle.tsx:9]: This is such a cool way to do it! Love that it's reusable.

File: src/components/Form/TourForm.tsx [hunk 2]
```diff
@@ -1,12 +1,79 @@
...
 
-export default function TourForm() {
   return (
     <div>
-      
+      <input
+        type='date'
+        value={state.date}
+        onChange={e => setState({ ...state, date: e.target.value})}
+        min={Date.now()}
```
Comment (tashiad) [inline src/components/Form/TourForm.tsx:62]: The min currently isn't working, right?

File: src/components/Form/Plan.tsx [hunk 3]
```diff
@@ -1,9 +1,39 @@
...
+import './Form.css'
+
+interface PlanProps {
+  renderTextInputs: (fields: string[]) => JSX.Element[],
+  isChecked: (fields: string[]) => boolean
+}
+
+
+export const Plan: React.FC<PlanProps> = ({ renderTextInputs, isChecked }) => {
+  const hazardFields = ['hazardWeather', 'hazardAvalanche', 'hazardSummary']
```
Comment (tashiad) [inline src/components/Form/Plan.tsx:14]: Nice

File: src/components/Form/TourForm.tsx [hunk 4]
```diff
@@ -1,12 +1,75 @@
...
+    hazardSummary: '',
+    routePreview: '',
+    routeAlternative: '',
+    emergencyPlan: '',
+    debriefConditions: '',
+    debriefDecisions: '',
+    debriefPlan: ''
+  })
+
+  const renderTextInputs = (fields: string[]): JSX.Element[] => {
```
Comment (rachelbuchta) [inline src/components/Form/TourForm.tsx:40]: This is such a cool way to keep things dynamic and not filling a component up with redundant input fields. Love it!

File: tsconfig.json [hunk 5]
```diff
@@ -18,7 +18,7 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react"
+    "jsx": "react-jsx"
```
Comment (tashiad) [inline tsconfig.json:21]: Really don't know why this is breaking on my machine only 🤷‍♀️

**Review by tashiad — APPROVED** [review]

File: src/components/Form/TourForm.tsx [hunk 6]
```diff
@@ -1,12 +1,75 @@
...
+    return fields.map(field=> {
+
+      return (
+        <TextField
+          value={state[field as keyof TourFormState]}
```
Comment (rachelbuchta) [inline src/components/Form/TourForm.tsx:45]: This is good to know how this works in TypeScript land too. Thanks for figuring that out.

File: src/components/Form/TourForm.tsx [hunk 7]
```diff
@@ -1,12 +1,75 @@
...
 
-export default function TourForm() {
   return (
     <div>
-      
+      <input
+        type='date'
+        value={state.date}
+        onChange={e => setState({ ...state, date: e.target.value})}
+        min={Date.now()}
```
Comment (rachelbuchta) [inline src/components/Form/TourForm.tsx:62]: Did this actually work or is it something we will have to go back to later?

**Review by rachelbuchta — APPROVED** [review]
Awesome job on making things dynamic and getting through those TypeScript challenges!

File: src/components/Form/TourForm.tsx (hunk 2 above)
Comment (deadbelly) [reply src/components/Form/TourForm.tsx:62]: No it's not. I wanted to leave the code there as a reminder to dig up solutions while working on that file.

File: src/components/Form/TourForm.tsx (hunk 7 above)
Comment (deadbelly) [reply src/components/Form/TourForm.tsx:62]: We need to come back to it later.

**Pull request merged**